drivers/w5500/w5500.c

```diff
@@ -0,0 +1,470 @@
+/*
+ * Copyright (C) 2023 Stefan Schmidt
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_w5500
+ * @{
+ *
+ * @file
+ * @brief       Device driver implementation for w5500 Ethernet devices
+ *
+ * @author      Stefan Schmidt <stemschmidt@gmail.com>
+ *
+ * @}
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <string.h>
+
+#include "assert.h"
+#include "iolist.h"
+#include "net/ethernet.h"
+#include "net/eui_provider.h"
+#include "net/netdev/eth.h"
+#include "net/ethernet/hdr.h"
+#include "ztimer.h"
+
+#include "w5500.h"
+#include "w5500_regs.h"
+
+#define ENABLE_DEBUG        0
+#include "debug.h"
+#include "log.h"
+
+/* Socket definitions. */
+#define BSB_SOCKET_BASE         (0x08)      /**< Base for Socket n registers. */
+#define RXBUF_16KB_TO_S0        (0x10)      /**< Receive memory size: 16 kB. */
+#define TXBUF_16KB_TO_S0        (0x10)      /**< Transmit memory size: 16 kB. */
+
+static const netdev_driver_t netdev_driver_w5500;
+
+static inline void send_addr(w5500_t *dev, uint16_t addr)
+{
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+    spi_transfer_byte(dev->p.spi, dev->p.cs, true, (addr >> 8));
+    spi_transfer_byte(dev->p.spi, dev->p.cs, true, (addr & 0xff));
+#else
+    spi_transfer_byte(dev->p.spi, dev->p.cs, true, (addr & 0xff));
+    spi_transfer_byte(dev->p.spi, dev->p.cs, true, (addr >> 8));
+#endif
+}
```

Comment:
This is actually incorrect. I think that https://github.com/RIOT-OS/RIOT/pull/20312 should be helpful. Otherwise:

```suggestion
static inline void send_addr(w5500_t *dev, uint16_t addr)
{
    spi_transfer_byte(dev->p.spi, dev->p.cs, true, (addr >> 8));
    spi_transfer_byte(dev->p.spi, dev->p.cs, true, (addr & 0xff));
}
```

(The shift operator is defined in regard to the numeric value, not in regard to the bit pattern it would yield in memory when stored in the host byte order. Hence, the operator behaves the same regardless of byte order.)